core/client/src/client.rs

```diff
@@ -1838,6 +1818,49 @@ impl<B, E, Block, RA> backend::AuxStore for Client<B, E, Block, RA>
 	}
 }
 
+
+impl<B, E, Block, RA> backend::AuxStore for &Client<B, E, Block, RA>
```

Comment:
Do we need this implementations? Isn't rust able to resolve this on its own?

Reply:
You would think that, wouldn't you? But no, unfortunately, it does not:

```
  Compiling subkey v2.0.0 (/home/ben/Dev/parity/substrate/subkey)
error[E0277]: the trait bound `&substrate_client::client::Client<B, E, Block, RA>: substrate_client::backend::AuxStore` is not satisfied
   --> core/finality-grandpa/src/light_import.rs:137:3
    |
137 |         do_import_block::<_, _, _, GrandpaJustification<Block>>(
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `substrate_client::backend::AuxStore` is not implemented for `&substrate_client::client::Client<B, E, Block, RA>`
    |
    = help: the following implementations were found:
              <substrate_client::client::Client<B, E, Block, RA> as substrate_client::backend::AuxStore>
note: required by `light_import::do_import_block`
   --> core/finality-grandpa/src/light_import.rs:234:1
    |
234 | / fn do_import_block<B, C, Block: BlockT<Hash=H256>, J>(
235 | |     mut client: C,
236 | |     data: &mut LightImportData<Block>,
237 | |     mut block: BlockImportParams<Block>,
...   |
289 | |     }
290 | | }
    | |_^

error[E0277]: the trait bound `&substrate_client::client::Client<B, E, Block, RA>: substrate_client::backend::AuxStore` is not satisfied
   --> core/finality-grandpa/src/light_import.rs:182:3
    |
182 |         do_import_finality_proof::<_, _, _, GrandpaJustification<Block>>(
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `substrate_client::backend::AuxStore` is not implemented for `&substrate_client::client::Client<B, E, Block, RA>`
    |
    = help: the following implementations were found:
              <substrate_client::client::Client<B, E, Block, RA> as substrate_client::backend::AuxStore>
note: required by `light_import::do_import_finality_proof`
   --> core/finality-grandpa/src/light_import.rs:293:1
    |
293 | / fn do_import_finality_proof<B, C, Block: BlockT<Hash=H256>, J>(
294 | |     client: C,
295 | |     backend: Arc<B>,
296 | |     authority_set_provider: &dyn AuthoritySetForFinalityChecker<Block>,
...   |
357 | |     Ok((finalized_block_hash, finalized_block_number))
358 | | }
    | |_^

   Compiling node-testing v2.0.0 (/home/ben/Dev/parity/substrate/node/testing)
error: aborting due to 2 previous errors
```